plugins/queue/scheduler.js

```diff
@@ -0,0 +1,614 @@
+        { jobId: config.jobId }
+    ]);
+
+    return executor.redisBreaker.runCommand('hdel', executor.periodicBuildTable, config.jobId);
+}
+
+/**
+ * Stops a previously enqueued frozen build in an executor
+ * @async  stopFrozen
+ * @param  {Object}  config        Configuration
+ * @param  {Integer} config.jobId  ID of the job with frozen builds
+ * @return {Promise}
+ */
+async function stopFrozen(executor, config) {
+    await executor.connect();
+
+    await executor.queueBreaker.runCommand('delDelayed', executor.frozenBuildQueue, 'startFrozen', [
+        { jobId: config.jobId }
+    ]);
+
+    return executor.redisBreaker.runCommand('hdel', executor.frozenBuildTable, config.jobId);
+}
+
+/**
+ * Calls postBuildEvent() with job configuration
+ * @async startFrozen
+ * @param {Object} config       Configuration
+ * @return {Promise}
+ */
+async function startFrozen(executor, config) {
+    try {
+        const newConfig = {
+            job: {
+                name: config.jobName
+            },
+            causeMessage: 'Started by freeze window scheduler'
+        };
+
+        if (config.jobState === 'DISABLED' || config.jobArchived === true) {
+            logger.error(`job ${config.jobName} is disabled or archived`);
+
+            return Promise.resolve();
+        }
+
+        Object.assign(newConfig, config);
+
+        return await postBuildEvent(executor, newConfig);
+    } catch (err) {
+        logger.error(`frozen builds: failed to post build event for job ${config.jobId}:${config.pipeline.id} ${err}`);
+
+        return Promise.resolve();
+    }
+}
+
+/**
+ * Starts a new periodic build in an executor
+ * @method startPeriodic
+ * @param {Object}   config              Configuration
+ * @param {Object}   config.pipeline     Pipeline of the job
+ * @param {Object}   config.job          Job object to create periodic builds for
+ * @param {String}   config.apiUri       Base URL of the Screwdriver API
+ * @param {Boolean}  config.isUpdate     Boolean to determine if updating existing periodic build
+ * @param {Boolean}  config.triggerBuild Flag to post new build event
+ * @return {Promise}
+ */
+async function startPeriodic(executor, config) {
+    const { pipeline, job, isUpdate, triggerBuild } = config;
+    // eslint-disable-next-line max-len
+    const buildCron = reach(job, 'permutations>0>annotations>screwdriver.cd/buildPeriodically', { separator: '>' });
+
+    if (isUpdate) {
+        await stopPeriodic(executor, { jobId: job.id });
+    }
+
+    if (triggerBuild) {
+        config.causeMessage = 'Started by periodic build scheduler';
+
+        // Even if post event failed for this event after retry, we should still enqueue the next event
+        try {
+            await postBuildEvent(executor, config);
+        } catch (err) {
+            logger.error(
+                `periodic builds: failed to post build event for job ${job.id} in pipeline ${pipeline.id}: ${err}`
```

Comment:
`err` can be the second argument for `logger.error`

Reply:
ok, sometimes logger.error doesn't escape the err object and prints garbage